Sourcery/Generating/Template/Swift/SwiftTemplate.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  SwiftTemplate.swift
+//  Sourcery
+//
+//  Created by Krunoslav Zaher on 12/30/16.
+//  Copyright © 2016 Pixle. All rights reserved.
+//
+
+import Foundation
+import PathKit
+
+enum SwiftTemplateParsingError: Error {
+    case unmatchedOpening(path: Path, line: Int)
+    case compilationError(path: Path, error: String)
+}
+
+enum Delimiters {
+    static let open = "<%"
+    static let close = "%>"
+}
+
+func generateSwiftCode(templateContent _templateContent: String, path: Path) throws -> String {
+    let templateContent = "<%%>" + _templateContent
+
+    let components = templateContent.components(separatedBy: Delimiters.open)
+
+    var sourceFile = [String]()
+
+    let line = {
+        return sourceFile.joined(separator: "").numberOfLineSeparators
+    }
+
+    for component in components.suffix(from: 1) {
+        guard let endIndex = component.range(of: Delimiters.close) else {
+            throw SwiftTemplateParsingError.unmatchedOpening(path: path, line: line())
+        }
+
+        let code = component.substring(to: endIndex.lowerBound)
+        if code.hasPrefix("=") {
+            let codeStartIndex = code.index(code.startIndex, offsetBy: 1)
+            let realCode = code.substring(from: codeStartIndex)
+            sourceFile.append("\n print(\"\\(" + realCode + ")\", terminator: \"\");")
+        } else {
+            sourceFile.append(code)
+        }
+
+        let encodedPart = component.substring(from: endIndex.upperBound)
+        sourceFile.append(("\n print(\"") + encodedPart.stringEncoded + "\", terminator: \"\");")
+        for _ in 0 ..< encodedPart.numberOfLineSeparators {
+            sourceFile.append("\n")
+        }
+    }
+
+    return sourceFile.joined(separator: "")
+}
+
+struct ProcessResult {
+    let output: String
+    let error: String
+}
+
+private extension Path {
```

Comment:
we could create a simple extension that both tests and sourcery can reuse, the code is the same outside of the path component which we can parametrize